### PR TITLE
Wagtail Transfer: main addition to codebase

### PR DESCRIPTION
**Pull request by stevejalim**
This PR adds the main bulk of the code for Wagtail Transfer to the codebase, but in a way that does NOT change default behaviour. (Partly to avoid developer pain and partly because we're waiting for a new release of WT)

Almost all of the code in this PR is NOT new - it's been reviewed as PRs for child tickets that were merged to this branch. 

I've marked any new code with Github comments, below.

## To do (delete all that do not apply):
 - [X] Ticket exists in Jira  https://uktrade.atlassian.net/browse/GP2-579 / https://uktrade.atlassian.net/browse/GP2-568 / https://uktrade.atlassian.net/browse/GP2-569  
 - [X] Jira ticket has the correct status.
 - [X] [Changelog](CHANGELOG.md) entry added.

 - [X] Manual testing done by @stevejalim 
 - [X] (if updating requirements) Requirements have been compiled.
 - [X] Added any new environment variable to vault.
 
 - [X] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: docs/wagtail_transfer.rst [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+* handle the moving of large files between systems without the risk of exceeding a 30-second timeout (for us on Gov UK PaaS).
+* automatically reflect the deletion or movemement of pages on a source site. Support can be added, but needs further implementation work. For now we have a manual workaround.
+* automatically handle the transfer of large media files that are featured in pages. This is more about the request timeout threshold of the platform we're on than W-T itself. Again, we can manually work around this.
+
+There are workarounds documented in Confluence for all of the above.
+
+
+GOTCHAS for developers
+----------------------
+
```
Comment (stevejalim) [inline docs/wagtail_transfer.rst:46]: @webbyfox @bobmeredith @depsiatwal @agiamas Can you all please read these minor gotchas, so you're aware of them. I'll do my best to minimise any pain that WT might cause you if you include me in code review for anything that adds new models

File: config/utils.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    config = {}
+
+    active_environment = env.str(ENV_IDENTIFICATION_KEY)
+
+    if active_environment == DEV:
+        # Dev needs to know about Staging and Beta to import FROM them
+        config.update({
+            BETA: {
+                'BASE_URL': env.str('WAGTAILTRANSFER_BASE_URL_BETA'),
```
Comment (depsiatwal) [inline config/utils.py:35]: do we need this level of flexibility ? 
Wondering if we can infer the URL based on a base one that's set less logic :-)

File: docs/wagtail_transfer.rst (hunk 1 above)
Comment (depsiatwal) [reply docs/wagtail_transfer.rst:46]: i'm lazy and don't want to read all this :-) 
I suggest a guinea pig to test the actual instructions

**Review by depsiatwal — COMMENTED** [review]
LGTM few comments

File: config/utils.py (hunk 2 above)
Comment (stevejalim) [reply config/utils.py:35]: Not really - some URLs are on private-DNS-only services, some are changing when we go live. I prefer explicit over magic/permuted

File: docs/wagtail_transfer.rst (hunk 1 above)
Comment (stevejalim) [reply docs/wagtail_transfer.rst:46]: I've done all this, but need y'all to be aware of it so we don't break things. Sometimes, we just gotta read the docs :) (and I know I don't always do this myself, it's true)

File: requirements_test.txt [hunk 3]
```diff
@@ -33,7 +34,7 @@ directory-constants==20.16.0  # via -r requirements.in, great-components
 directory-forms-api-client==5.0  # via -r requirements.in
 directory-sso-api-client==6.5.3  # via -r requirements.in
 directory-validators==6.0.3  # via -r requirements.in
-django-debug-toolbar==2.2  # via -r requirements_test.in
+django-debug-toolbar==3.1.1  # via -r requirements_test.in
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply requirements_test.txt:37]: just test requirements so I wouldn't be too concerned

File: config/utils.py (hunk 2 above)
Comment (depsiatwal) [reply config/utils.py:35]: fairpoint :-)

**Review by depsiatwal — APPROVED** [review]
LGTM

**Pull request merged**